### ISSUE TITLE
the model does not detect the LP if the picture was only the LP 

### ISSUE THREAD

**Issue by Fahad-Alsabr** [sergiomsilva/alpr-unconstrained#142]
hello everyone i tried to use photos of the LP only without the vehicle and the model doesn't extract the text of the LP , so i notes that if it was a photo of a vehicle with the  LP it will be detected otherwise it's not

**Comment by nghigd9** [comment]
try to change the DMIN from 288 to lower value (144, 112, ...) in file "license-plate-detection.py" line 42.

**Comment by LeeWlving** [comment]
Yeah! In the paper, the model will first detect the vehicle ,than detect the LP, so maybe if there is no vehicle in you photo , this model won't work well.

**Comment by LeeWlving** [comment]
> hello everyone i tried to use photos of the LP only without the vehicle and the model doesn't extract the text of the LP , so i notes that if it was a photo of a vehicle with the LP it will be detected otherwise it's not

Yeah! In the paper, the model will first detect the vehicle ,than detect the LP, so maybe if there is no vehicle in you photo , this model won't work well.

**Comment by ivanh23** [comment]
Actually , it does not need a vehicle just use the detect license plate script but the trained model that they shared is not the best , if you train your own it will detect the license plate , if you have any questions feel free to ask , i will try to help you

**Comment by yazeedhasan97** [comment]
Hello Guys,
do you have an idea about how to extract the OCR of this project only, without extracting the darknet or build it?

**Comment by Fahad-Alsabr** [comment]
> Actually , it does not need a vehicle just use the detect ocr script but the trained model that they shared is not the best , if you train your own it will detect the license plate or the ocr , if you have any questions feel free to ask , i will try to help you

thank you very much , i'm actually working on the LP of Saudi Arabia and i trained my model for about 70K iteration on google colab  because my computer is old so i used there GPU but after 70K iteration i got over usage of GPU in google colab . however when i run the model i only got 1 LP of saudi arabia is correct the rest of them either i got some of the letters or i got some of the numbers and this is my output , the only correct Saudi LP is the 4.1.jpg 

 `Scanning /tmp/output/03016_0car_lp.png
		LP: MPE3389
	Scanning /tmp/output/03025_0car_lp.png
		LP: INS6012
	Scanning /tmp/output/03033_0car_lp.png
		LP: SEZ229
	Scanning /tmp/output/03058_0car_lp.png
		LP: 1
	Scanning /tmp/output/03058_1car_lp.png
		LP: C2LJBH
	Scanning /tmp/output/03066_2car_lp.png
		LP: HHP8586
	Scanning /tmp/output/03071_0car_lp.png
		LP: 6GQR959
	Scanning /tmp/output/1.13_0car_lp.png
		LP: J7L
	Scanning /tmp/output/2.19_0car_lp.png
		LP: RDJ
	Scanning /tmp/output/4.1_0car_lp.png
		LP: 5622ZVB
	Scanning /tmp/output/4.2_0car_lp.png
		LP: 19143A
	Scanning /tmp/output/46_0car_lp.png
		LP: MH20CS9817`

**Comment by yazeedhasan97** [comment]
Hi there,
I could detect the plate using different model, however, I used the your ocr and found good results, still, there is a problem, the detected plate need a deblure so the chars and numbers become more clearer and accurate

I really hope that you could help me in finding a great deblure model that could also work in a realtime manner

**Comment by ivanh23** [comment]
Hmm, this can mean that 

> > Actually , it does not need a vehicle just use the detect ocr script but the trained model that they shared is not the best , if you train your own it will detect the license plate or the ocr , if you have any questions feel free to ask , i will try to help you
> 
> thank you very much , i'm actually working on the LP of Saudi Arabia and i trained my model for about 70K iteration on google colab because my computer is old so i used there GPU but after 70K iteration i got over usage of GPU in google colab . however when i run the model i only got 1 LP of saudi arabia is correct the rest of them either i got some of the letters or i got some of the numbers and this is my output , the only correct Saudi LP is the 4.1.jpg
> 
> `Scanning /tmp/output/03016_0car_lp.png LP: MPE3389 Scanning /tmp/output/03025_0car_lp.png LP: INS6012 Scanning /tmp/output/03033_0car_lp.png LP: SEZ229 Scanning /tmp/output/03058_0car_lp.png LP: 1 Scanning /tmp/output/03058_1car_lp.png LP: C2LJBH Scanning /tmp/output/03066_2car_lp.png LP: HHP8586 Scanning /tmp/output/03071_0car_lp.png LP: 6GQR959 Scanning /tmp/output/1.13_0car_lp.png LP: J7L Scanning /tmp/output/2.19_0car_lp.png LP: RDJ Scanning /tmp/output/4.1_0car_lp.png LP: 5622ZVB Scanning /tmp/output/4.2_0car_lp.png LP: 19143A Scanning /tmp/output/46_0car_lp.png LP: MH20CS9817`


Hmm i havent used google colab , but i you are detecting only couple of letters maybe that means that the training did not go well , try again with a lower batch size and iterations in the config file  , and one more thing , are you training YOLOv3 model or YOLOv2? because the ocr script cant work with YOLOv3 it gives false results and incomplete , if you are using YOLOv3 i have scipt that can help you .
PS how many images are you training on ?

**Comment by Fahad-Alsabr** [comment]
> Hmm, this can mean that
> 
> > > Actually , it does not need a vehicle just use the detect ocr script but the trained model that they shared is not the best , if you train your own it will detect the license plate or the ocr , if you have any questions feel free to ask , i will try to help you
> > 
> > 
> > thank you very much , i'm actually working on the LP of Saudi Arabia and i trained my model for about 70K iteration on google colab because my computer is old so i used there GPU but after 70K iteration i got over usage of GPU in google colab . however when i run the model i only got 1 LP of saudi arabia is correct the rest of them either i got some of the letters or i got some of the numbers and this is my output , the only correct Saudi LP is the 4.1.jpg
> > `Scanning /tmp/output/03016_0car_lp.png LP: MPE3389 Scanning /tmp/output/03025_0car_lp.png LP: INS6012 Scanning /tmp/output/03033_0car_lp.png LP: SEZ229 Scanning /tmp/output/03058_0car_lp.png LP: 1 Scanning /tmp/output/03058_1car_lp.png LP: C2LJBH Scanning /tmp/output/03066_2car_lp.png LP: HHP8586 Scanning /tmp/output/03071_0car_lp.png LP: 6GQR959 Scanning /tmp/output/1.13_0car_lp.png LP: J7L Scanning /tmp/output/2.19_0car_lp.png LP: RDJ Scanning /tmp/output/4.1_0car_lp.png LP: 5622ZVB Scanning /tmp/output/4.2_0car_lp.png LP: 19143A Scanning /tmp/output/46_0car_lp.png LP: MH20CS9817`
> 
> Hmm i havent used google colab , but i you are detecting only couple of letters maybe that means that the training did not go well , try again with a lower batch size and iterations in the config file , and one more thing , are you training YOLOv3 model or YOLOv2? because the ocr script cant work with YOLOv3 it gives false results and incomplete , if you are using YOLOv3 i have scipt that can help you .
> PS how many images are you training on ?

i used google colab because it better and easy to use with GPU . i used YOLOv2 , i triad with YOLOv3 but i got nothing  .  the Saudi LP  is two layer the first in Arabic and the second in English but i only need the one in English you may google it so you can understand it better  .and for the training i used 98 images with annotation Xml file , and for the annotation  i used website called Nanonets.com to label and to download the annotation

**Comment by Fahad-Alsabr** [comment]
maybe if i can used the available model by the developer to detect the whole  second layer only i may have a better results .but the problem with Saudi LP is that it has many shapes the long rectangle LP and the short rectangle  and some of the long rectangle LP has a word  " KSA " written in the middle of the LP

**Comment by shubham-shahh** [comment]
> Actually , it does not need a vehicle just use the detect ocr script but the trained model that they shared is not the best , if you train your own it will detect the license plate or the ocr , if you have any questions feel free to ask , i will try to help you

Have you trained your own?
